File: kotlin/lang/security/use-of-sha1.kt
import java.security.MessageDigest
import org.apache.commons.codec.digest.DigestUtils

public class WeakHashes {
  public fun sha1(password: String): Array<Byte> {
      // ruleid: use-of-sha1
      var sha1Digest: MessageDigest = MessageDigest.getInstance("SHA1")
      sha1Digest.update(password.getBytes())
      val hashValue: Array<Byte> = sha1Digest.digest()
      return hashValue
  }
  public fun sha1b(password: String): Array<Byte> {
      // ruleid: use-of-sha1
      var sha1Digest: MessageDigest = MessageDigest.getInstance("SHA-1")
      sha1Digest.update(password.getBytes())
      val hashValue: Array<Byte> = sha1Digest.digest()
      return hashValue
  }
  public fun sha1_digestutil(password: String): Array<Byte> {
    // ruleid: use-of-sha1
    val hashValue: Array<Byte> = DigestUtils.getSha1Digest().digest(password.getBytes())
    return hashValue
  }
}

File: clojure/lang/security/use-of-sha1.clj
(import 'java.security.MessageDigest
        'java.math.BigInteger)

(defn sha1 [s]
  // ruleid: use-of-sha1
  (let [algorithm (MessageDigest/getInstance "SHA-1")
        size (* 2 (.getDigestLength algorithm))
        raw (.digest algorithm (.getBytes s))
        sig (.toString (BigInteger. 1 raw) 16)
        padding (apply str (repeat (- size (count sig)) "0"))]
    (str padding sig)))

(defn sha1b [s]
  // ruleid: use-of-sha1
  (let [algorithm (MessageDigest/getInstance MessageDigestAlgorithms/SHA-1)
        size (* 2 (.getDigestLength algorithm))
        raw (.digest algorithm (.getBytes s))
        sig (.toString (BigInteger. 1 raw) 16)
        padding (apply str (repeat (- size (count sig)) "0"))]
    (str padding sig)))

(defn sha1c [s]
  // ruleid: use-of-sha1
  (let [algorithm (MessageDigest/getInstance org.apache.commons.codec.digest.MessageDigestAlgorithms/SHA-1)
        size (* 2 (.getDigestLength algorithm))
        raw (.digest algorithm (.getBytes s))
        sig (.toString (BigInteger. 1 raw) 16)
        padding (apply str (repeat (- size (count sig)) "0"))]
    (str padding sig)))

(defn sha256 [s]
  // ok: use-of-sha1
  (let [algorithm (MessageDigest/getInstance "SHA-256")
        size (* 2 (.getDigestLength algorithm))
        raw (.digest algorithm (.getBytes s))
        sig (.toString (BigInteger. 1 raw) 16)
        padding (apply str (repeat (- size (count sig)) "0"))]
    (str padding sig)))

(defn sha256b [s]
  // ok: use-of-sha1
  (let [algorithm (MessageDigest/getInstance MessageDigestAlgorithms/SHA-256)
        size (* 2 (.getDigestLength algorithm))
        raw (.digest algorithm (.getBytes s))
        sig (.toString (BigInteger. 1 raw) 16)
        padding (apply str (repeat (- size (count sig)) "0"))]
    (str padding sig)))

(defn sha256c [s]
  // ok: use-of-sha1
  (let [algorithm (MessageDigest/getInstance org.apache.commons.codec.digest.MessageDigestAlgorithms/SHA-256)
        size (* 2 (.getDigestLength algorithm))
        raw (.digest algorithm (.getBytes s))
        sig (.toString (BigInteger. 1 raw) 16)
        padding (apply str (repeat (- size (count sig)) "0"))]
    (str padding sig)))
